src/Language/PureScript/Errors.hs

```diff
@@ -639,29 +642,31 @@ prettyPrintSingleError full level e = prettyPrintErrorMessage . positionHintsFir
             , line "Consider moving the instance, if possible, or using a newtype wrapper."
             ]
     renderSimpleErrorMessage InvalidNewtype =
-      line "Newtypes must define a single constructor with a single argument"
+      line "Newtypes must define a single constructor with a single argument."
     renderSimpleErrorMessage (InvalidInstanceHead ty) =
-      paras [ line "Invalid type in class instance head:"
+      paras [ line "An invalid type appears in a type class instance head:"
             , indent $ typeAsBox ty
             ]
```

Comment:
Can we explain this further? Like actually enumerate the rules for what is considered to be an acceptable type in an instance head. Also, this error makes it sound like the type itself is somehow invalid rather than the instance head...

"Type class instance head is invalid due to the use/appearance/... of type x`?